### PR TITLE
Various fixes

### PR DESCRIPTION
**Pull request by pulsejet**
~Not for review yet - will also need tag on go-ndn~

### CODE REVIEW

**Comment by zjkmxy** [comment]
What's the reason for copying the frame?

**Comment by pulsejet** [comment]
> What's the reason for copying the frame?

The ndnlp link service uses a single outgoing buffer that is passed to the downstream transport (to save mallocgc; I profiled this and the overhead was significant ~5%). The downstream transport must synchronously copy the buffer to OS memory.

https://github.com/named-data/YaNFD/blob/62198cd97f861ecf5439be6a03e18b6261804b91/face/ndnlp-link-service.go#L269

In this case it loops back to us, so we need to copy it ourselves.

File: fw/strategy.go [hunk 1]
```diff
@@ -44,14 +54,17 @@ type StrategyBase struct {
...
+	version uint64,
+	strategyLogName string,
 ) {
 	var err error
 	s.thread = fwThread
 	s.threadID = s.thread.threadID
 	s.name, err = enc.NameFromStr(StrategyPrefix)
 	if err != nil {
-		core.LogFatal(s, "StrategyPrefix is a bad NDN Name")
+		panic("StrategyPrefix is a bad NDN Name")
```
Comment (zjkmxy) [inline fw/strategy.go:67]: Log should be used whenever possible. If necessary let's put it in a separate package.

**Review by zjkmxy — COMMENTED** [review]
LGTM till now

File: fw/strategy.go (hunk 1 above)
Comment (pulsejet) [reply fw/strategy.go:67]: I feel for fatal logs panic is more appropriate to signal the developers intention (I don't "know" if log fatal would crash).

**Comment by pulsejet** [comment]
Thanks for taking a look @zjkmxy , I was worried the changes are a bit too wild to review easily xD

Almost done now for this PR, just need a new tag on go-ndn. Next PR I'll check/fix the other transports (likely only unix works at the moment)

File: fw/strategy.go (hunk 1 above)
Comment (zjkmxy) [reply fw/strategy.go:67]: `LogFatal` will call `panic` and kill the program.

**Comment by zjkmxy** [comment]
> will also need tag on go-ndn

Tagged as [`v0.0.8`](https://pkg.go.dev/github.com/zjkmxy/go-ndn@v0.0.8)

File: fw/strategy.go (hunk 1 above)
Comment (pulsejet) [reply fw/strategy.go:67]: Yes, but visually it's more obvious to the developer when you see a panic -- means a crash
Also tooling is better at figuring that out

**Review by zjkmxy — APPROVED** [review]

**Pull request merged**